### ISSUE TITLE
erroneous message from `chebfun(@abs,'trig')`

### ISSUE THREAD

**Issue by trefethen** [chebfun/chebfun#1514]
This comes up in revising the trigfun paper for _SISC_.  In the following there's an mismatch between 65536 and 65537:
```
>> chebfun(@abs,'trig')
Warning: Function not resolved using 65537 pts. Have you tried a non-trig representation? 
ans =
   chebfun column (1 smooth piece)
       interval       length     endpoint values trig
[      -1,       1]    65536         1        1 
vscale = 1.
```
Can @mhsnjvd or @gradywright or @Hadrien-Montanelli fix this message appropriately?  But in fact there's also a question of a factor of 2 here.  The actual vectors at which such a command evaluates the function turn out to have lengths
```
 2 1 16 1 16 32 64 128 256 512 1024 2048 4096 8192 16384 32768
```
but not 65536.  What's going on?

**Comment by aaustin141** [comment]
I can answer the last one:  the reason sampling appears to stop at 32,678 instead of 65,536 is due to nesting of the grids; see #1337.

**Comment by gradywright** [comment]
I started to work out a fix for #1337 by first fixing #1336.  However, I got stuck on how best to fix #1336 given issues with `chebfunpref`.  Please see the discussion in #1336.  @aaustin141 any ideas?

**Comment by aaustin141** [comment]
@gradywright:  Thanks for reminding me that we'd visited this issue before in #1336.  See #1520.  I've implemented my suggestion from #1336 to merge the tech preferences "on-demand" whenever we need them.  I didn't notice any slowdown from doing this, so I guess we're set.  Once those changes get merged, you should be able to set the default value for `maxLength` in `@trigtech/techPrefs.m` as you once tried and have it honored properly this time.

**Comment by aaustin141** [comment]
@gradywright:  @nickhale merged #1520 a short while ago, so you should be good to go.  Let me know if you run into any other issues.

**Comment by gradywright** [comment]
Thanks @aaustin141 (and @nickhale).  I will start working on this today.

**Comment by aaustin141** [comment]
Closed in #1523.
